user.go

```diff
@@ -39,6 +42,58 @@ func (a *App) GetUser(userID string) (*User, error) {
 	}
 }
 
+// GetUserByIdentifier gets a user using their identifier
+// returns user on success, error on failure
+func (a *App) GetUserByIdentifier(identifier string) (*User, error) {
+	var errorText string
+	message := "failed to get Passage User By Identifier"
+	limit := 1
+	res, err := a.client.ListPaginatedUsersWithResponse(
+		context.Background(),
+		a.ID,
+		&ListPaginatedUsersParams{
+			Limit:      &limit,
+			Identifier: &identifier,
+		},
+	)
+
+	if err != nil {
+		return nil, Error{Message: "network error:failed to get Passage User by Identifier"}
+	}
+
+	if res.JSON200 != nil {
+		users := res.JSON200.Users
+		if len(users) == 0 {
+			message = fmt.Sprintf(IdentifierDoesNotExist, identifier)
+			return nil, Error{
+				Message:    message,
+				StatusCode: 404,
+				StatusText: "404 not found",
+				ErrorText:  "user_not_found",
+			}
+		}
+
+		return a.GetUser(users[0].ID)
```

Comment:
We may want to handle the case where `len(users) > 1` ([example](https://github.com/passageidentity/backend-services/blob/main/identity-service/interactors/user/user.go#L137-L140))

Reply:
users will never be greater than 1 since we are setting the limit on the pagination to be `1`.

Reply:
Oh duh haha sorry bout that